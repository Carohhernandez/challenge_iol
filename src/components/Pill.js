import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const PillContainer = styled.div`
    display: flex;
    align-items: center;

    border-radius: 50px;
    background-color: #abb2bf;

    padding: 10px;
`;

const PillThumbContainer = styled.div`
    width: 20%;
    display: flex;
    justify-content: flex-start;
    align-items: center;
`;

const PillThumb = styled.img`
    width: 100%;
    height: 70px;

    border-radius: 50%;
`;

const PillInformation = styled.div`
    width: 80%;
    display: flex;
    flex-direction: column;

    padding-left: 10px;
`;

const PillTitle = styled(Link)`
    text-decoration: none;
    color: #fff;

    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
`;

const PillDescription = styled.p`
    display: inline-block;

    margin: 5px 0px;
    font-size: 14px;
    font-style: italic;

    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;

    i {
        margin-right: 5px;
    }
`;

const Pill = ({item, url}) => {
    return (
        <PillContainer key={item?.id}>
            <PillThumbContainer>
                <PillThumb src={item?.image} alt={item?.name} />
            </PillThumbContainer>
            <PillInformation>
                <PillTitle to={url}>
                    {item?.name}
                </PillTitle>
                <PillDescription>
                    <i className="fa fa-map-marker" aria-hidden="true"></i>
                    {item?.location?.name}
                </PillDescription>
            </PillInformation>
        </PillContainer>
    );
};

export default Pill;