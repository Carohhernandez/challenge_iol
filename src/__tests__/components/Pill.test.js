import { render, cleanup } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Pill from '../../components/Pill';

const item = {
    "id":10,
    "name":"Alan Rails",
    "status":"Dead",
    "species":"Human",
    "type":"Superhuman (Ghost trains summoner)",
    "gender":"Male",
    "origin":{
        "name":"unknown",
        "url":""
    },
    "location":{
        "name":"Worldender's lair",
        "url":"https://rickandmortyapi.com/api/location/4"
    },
    "image":"https://rickandmortyapi.com/api/character/avatar/10.jpeg",
    "episode":["https://rickandmortyapi.com/api/episode/25"],
    "url":"https://rickandmortyapi.com/api/character/10",
    "created":"2017-11-04T20:19:09.017Z"
};

const url = '/characters/10';

afterEach(cleanup);

test("renders Pill correctly", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <Pill item={item} url={url} />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});