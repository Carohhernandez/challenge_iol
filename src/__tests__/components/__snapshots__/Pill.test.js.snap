// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pill correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe eeQyYs"
  >
    <div
      class="sc-gswNZR fKhFco"
    >
      <img
        alt="Alan Rails"
        class="sc-dkrFOg gOYJRp"
        src="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
      />
    </div>
    <div
      class="sc-hLBbgP fYoeVE"
    >
      <a
        class="sc-eDvSVe jlcEpK"
        href="/characters/10"
      >
        Alan Rails
      </a>
      <p
        class="sc-jSUZER hsUrVP"
      >
        <i
          aria-hidden="true"
          class="fa fa-map-marker"
        />
        Worldender's lair
      </p>
    </div>
  </div>
</DocumentFragment>
`;
